ASTInfoGenerator/Specifiers/SpecifierManager.py

```diff
@@ -0,0 +1,72 @@
+from Specifiers.Specifier import Specifier
+from Specifiers.FunctionStructure.Function import Function
+from Specifiers.StructStructure.Struct import Struct
+from ASTInfoGenerator.Debug import *
+
+
+class SpecifierManager:
+    """Manages a list of specifiers
+    """
+
+    def __init__(self):
+        self.__specifiers: list[Specifier] = []
+
+    # if function is not found, a new one is returned with name function_name
+    def get_function(self, function_name: str) -> Function:
+        for specifier in self.__specifiers:
+            if type(specifier) is Function:
+                if specifier.equals(function_name):
+                    return specifier
+
+        new_function = Function(function_name, None)
+        self.__specifiers.append(new_function)
+        return self.__specifiers[-1]
+
+    # if struct is not found, a new one is returned with name struct_name if error_if_not_found is false
+    def get_struct(self, struct_name: str, error_if_not_found: bool = False) -> Struct:
+        for specifier in self.__specifiers:
+            if type(specifier) is Struct:
+                if specifier.equals(struct_name):
+                    return specifier
+
+        if error_if_not_found:
+            raise ValueError(f"struct {struct_name} not found")
+
+        new_struct = Struct(struct_name)
+        self.__specifiers.append(new_struct)
+        return self.__specifiers[-1]
+
+    # if there already exists a function of name function_name, that function is returned.
+    # otherwise, the function is added and returned.
```

Comment:
There doesn't seem to be much of a difference between `get_function` and `add_function` (and the same is true for the struct equivalents): get adds the function if it isn't found, and add gets the function if it is found. Can you combine the implementations?

Reply:
`get_function` was actually unused; deleting. I'll combine `get_struct` and `add_struct` into `get_or_add_struct`. The feature to error if not found was never actually used, so that parameter will be removed.